.github/workflows/base.yml

```diff
@@ -0,0 +1,77 @@
+name: check
+on: [push, pull_request]
+jobs:
+  # The build and test job builds librdkafka, a dynamically linked version of
+  # the go client, and then runs all the tests which don't require a broker.
+  # We also build the documentation (which also tests out error generation).
+  # We can't use prebuilt librdkafka since it might not contain changes from
+  # master, and so, we have to keep the build, test, and docs steps in one job
+  # to avoid needing to build librdkafka repeatedly.
+  build-test-doc:
+    runs-on: ubuntu-latest
+    steps:
+      # Setup repositories and dependencies for confluent-kafka-go
+      - uses: actions/checkout@v3 # checks out self-repo by default
+      - uses: actions/checkout@v3
```

Comment:
why is this duplicated?

Reply:
the first one checks out the repo (confluent-kafka-go), the second one checks out librdkafka. I'll make the comment more clear.